indexing-service/src/main/java/io/druid/indexing/common/actions/RemoteTaskActionClient.java

```diff
@@ -155,7 +155,7 @@ private long jitter(long input){
 
   private URI makeServiceUri(final Server instance) throws URISyntaxException
   {
-    return new URI(String.format("%s://%s%s", instance.getScheme(), instance.getHost(), "/druid/indexer/v1/action"));
+    return new URI(instance.getScheme(), null, instance.getAddress(), instance.getPort(), "/druid/indexer/v1/action", null, null);
```

Comment:
no need for the null values here, I think, based on how you changed in `IndexingServiceClient.java`